src/test/java/seedu/address/logic/parser/FindCommandParserTest.java

```diff
@@ -20,15 +26,47 @@ public void parse_emptyArg_throwsParseException() {
         assertParseFailure(parser, "     ", String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
     }
 
+    @Test
+    public void parse_moreThanOnePrefix_throwsParseException() {
+        assertParseFailure(parser, " n/CS1231S Homework mod/CS2100", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MORE_THAN_ONE_PREFIX_MESSAGE));
+    }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        //invalid prefix
+        assertParseFailure(parser, " boo/CS2100", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        //invalid keywords depending on prefix
+        assertParseFailure(parser, " d/2345 hello", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.INVALID_DATE_OR_TIME_MESSAGE));
+        assertParseFailure(parser, " mod/CS386", ModuleCode.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, " priority/NO", Priority.MESSAGE_CONSTRAINTS);
+        //preamble present
+        assertParseFailure(parser, " preamble mod/CS2100", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, " preamble n/assignment 2", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, " preamble d/1200", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, " preamble priority/MEDIUM", String.format(
+                MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+    }
```

Comment:
Nice test cases :) A suggestion is inserting white spaces between each test case to improve readability. You may want to consider testing for incorrect date and time as well (i.e 31-02-2020 or 2500) to ensure that the parser works as intended :)

Reply:
Amended!